atomic_reactor/utils/resource_manager.py

```diff
@@ -0,0 +1,132 @@
+                        prid = slot_data.get("prid", "unknown")
+                    logger.debug("%s slot %s is %s", meta['hostname'], i, state)
+                    meta['slots'].append({'sid': i, 'state': state, 'prid': prid})
+                    i += 1
+                meta['free_slots'] = free_slots
+                return meta
+        except Exception as err:
+            logger.debug(err)
+            logger.error("Availiability check failed fo host: %s", host)
+            return False
+
+    def unlock(self, user, host, slot):
+        logger.debug('Unlocking')
+        retries = 0
+        while retries < 3:
+            cmd = ('ssh %s@%s -C4 "echo \'0\' > %sslot_%s.json"', user, host, ROOT_PATH, slot)
+            p = subprocess.Popen(cmd, shell=True)
+            ret = p.wait()
+            if ret == 0:
+                logger.info('Unlocked')
+                return True
+            else:
+                logger.debug('Unlocking failed, retrying')
+                retries += 1
+        return False
+
+    def lock(self, user, host, slot, prid):
+        logger.debug("Aquiring lock")
+        payload = json.dumps({"prid": prid, "locked": str(datetime.now())})
+        retries = 0
+        while retries < 3:
+            filename = os.path.join(ROOT_PATH, f'slot_{slot}.json')
+            _cmd = ('if grep -qw 0 %s ; then echo %s > %s ;else exit 1; fi',
+                    filename,
+                    quote(payload),
+                    filename
+                    )
+            cmd = 'flock {} -c {}'.format(filename, quote(_cmd))
+            remote_cmd = 'ssh {}@{} -C4 {}'.format(user, host, quote(cmd))
+            logger.debug(cmd)
+            p = subprocess.Popen(remote_cmd, shell=True, universal_newlines=True)
+            ret = p.wait()
+            if ret == 0:
+                logger.info('Locked')
+                return True
+            else:
+                logger.debug('Locking failed, retrying')
+                retries += 1
+        logger.info('Locking failed')
+        return False
+
+    def find(self, arch, hosts):
+        try:
+            with open(hosts, "r") as f:
+                hosts_list = json.loads(f)
+            resources = []
+            for host in hosts_list['hosts']:
+                user = host.get('username', False)
+                hostname = host.get('hostname', False)
+                if user and hostname:
+                    data = self.check_avail(user, hostname)
+                    if arch == data.get('arch', ''):
+                        if data.get('free_slots', 0) > 0:
+                            resources.append(data)
+                else:
+                    logger.warning('in hosts file is missing user gor host %s', hostname)
+            result = sorted(resources, key=lambda d: d['free_slots'])
+            logger.debug('sorted result: %s', result)
+            if len(result) > 0:
+                return result[0]
+            else:
+                return False
```

Comment:
So what's rtype here? dict or boolean?

Can we just return expected value or exception, pls try to avoid procedural check for results. Exceptions are preferred